pkgs/development/tools/google-java-format/default.nix

```diff
@@ -17,11 +17,14 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/{bin,share/google-java-format}
-    install -D ${src} $out/share/google-java-format/google-java-format.jar
-
     makeWrapper ${jre}/bin/java $out/bin/google-java-format \
-      --add-flags "-jar $out/share/google-java-format/google-java-format.jar"
+        --argv0 ${pname} \
+        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED" \
+        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED" \
+        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED" \
+        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED" \
+        --add-flags "--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED" \
+        --add-flags "-jar ${src}"
```

Comment:
Ah, I didn't mean that you should remove the `install` command. The flag was ok as before `-jar $out/share/google-java-format/google-java-format.jar` with the installed jar. `${src}` is no part of the derivation output. It needs to be installed.

Reply:
I saw that being done in `pkgs/games/unciv/default.nix`... Would using `src` like this end up breaking due to garbage collection then?

Reply:
Restored the `install` command. I'm curious about the other approach, though.

Reply:
That's probably not right. And those are few, most packages use `$out`:

``` console
$ rg -- '--add-flags.*-jar.*\$out' | wc -l
      76
$ rg -- '--add-flags.*-jar.*\$\{?src' | wc -l
       9
```